posthog/tasks/exports/csv_exporter.py

```diff
@@ -167,52 +168,57 @@ class UnexpectedEmptyJsonResponse(Exception):
 def _export_to_csv(exported_asset: ExportedAsset, limit: int = 1000, max_limit: int = 3_500) -> None:
     resource = exported_asset.export_context
 
-    path: str = resource["path"]
     columns: List[str] = resource.get("columns", [])
 
-    method: str = resource.get("method", "GET")
-    body = resource.get("body", None)
-
-    access_token = encode_jwt(
-        {"id": exported_asset.created_by_id}, datetime.timedelta(minutes=15), PosthogJwtAudience.IMPERSONATED_USER
-    )
-
-    next_url = None
     all_csv_rows: List[Any] = []
 
-    while len(all_csv_rows) < max_limit:
-        response = make_api_call(access_token, body, limit, method, next_url, path)
-
-        if response.status_code != 200:
-            # noinspection PyBroadException
-            try:
-                response_json = response.json()
-            except Exception:
-                response_json = "no response json to parse"
-            raise Exception(f"export API call failed with status_code: {response.status_code}. {response_json}")
-
-        # Figure out how to handle funnel polling....
-        data = response.json()
+    if resource.get("source"):
+        query = resource.get("source")
+        data = process_query(team=exported_asset.team, query_json=query, is_hogql_enabled=True)
+        all_csv_rows = _convert_response_to_csv_data(data)
```

Comment:
Long term this is probably what we want to do anyway tbh. If every exportable thing is representable as a hogql query... But thats an aside